crates/stdarch-gen/src/main.rs

```diff
@@ -0,0 +1,502 @@
+                        if link_aarch64.is_some() {
+                            panic!("[{}] Can't specify link and fn at the same time.", name)
+                        }
+                        current_fn
+                    } else {
+                        if link_aarch64.is_none() {
+                            panic!("[{}] Either fn or link-aarch have to be specified.", name)
+                        }
+                        format!("{}_", name)
+                    };
+                    let current_aarch64 = current_aarch64.clone().unwrap();
+                    let link = if let Some(link_aarch64) = link_aarch64.clone() {
+                        let ext = type_to_ext(in_t);
+
+                        format!(
+                            r#"
+    #[allow(improper_ctypes)]
+    extern "C" {{
+        #[cfg_attr(target_arch = "aarch64", link_name = "llvm.aarch64.neon.{}")]
+        fn {}(a: {}, a: {}) -> {};
+    }}
+"#,
+                            link_aarch64.replace("_EXT_", ext),
+                            current_fn,
+                            in_t,
+                            in_t,
+                            out_t
+                        )
+                    } else {
+                        String::new()
+                    };
+                    let function = format!(
+                        r#"
+{}
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr({}))]
+pub unsafe fn {}(a: {}, b: {}) -> {} {{
+    {}{}(a, b)
+}}
+"#,
+                        current_comment, current_aarch64, name, in_t, in_t, out_t, link, current_fn,
+                    );
+                    let test = format!(
+                        r#"
+    #[simd_test(enable = "neon")]
+    unsafe fn test_{}() {{
+        let a{};
+        let b{};
+        let e{};
+        let r: {} = transmute({}(transmute(a), transmute(b)));
+        assert_eq!(r, e);
+    }}
+"#,
+                        name,
+                        values(globla_t, &a),
+                        values(globla_t, &b),
+                        values(globla_ret_t, &e),
+                        globla_ret_t,
+                        name
+                    );
+                    // out_aarch64.push_str(&link);
+                    out_aarch64.push_str(&function);
+                    tests_aarch64.push_str(&test);
+                }
+            }
+        }
+    }
+    tests_arm.push('}');
+    tests_arm.push('\n');
+    tests_aarch64.push('}');
+    tests_aarch64.push('\n');
+
+    let mut file_arm = File::create(ARM_OUT)?;
+    file_arm.write_all(out_arm.as_bytes())?;
+    file_arm.write_all(tests_arm.as_bytes())?;
+
+    let mut file_aarch = File::create(AARCH64_OUT)?;
+    file_aarch.write_all(out_aarch64.as_bytes())?;
+    file_aarch.write_all(tests_aarch64.as_bytes())?;
+    Command::new("sh")
+            .arg("-c")
+            .arg(format!("rustfmt {} {}", ARM_OUT, AARCH64_OUT))
```

Comment:
Please don't use `sh` as command to execute. Windows doesnt support it for example.